contracts/axelarnet-gateway/src/clients/external.rs

```diff
@@ -11,36 +12,35 @@ pub struct AxelarExecutableMsg {
     pub payload: HexBinary,
 }
 
-/// Crate-specific `ExecuteMsg` type wraps the `AxelarExecutableMsg` for the AxelarExecutable client.
-#[cw_serde]
-pub(crate) enum AxelarExecutableExecuteMsg {
-    /// Execute the message at the destination contract with the corresponding payload.
-    Execute(AxelarExecutableMsg),
+pub struct PayloadExecutor<'a> {
+    client: client::Client<'a, AxelarExecutableMsg, ()>,
 }
 
-impl<'a> From<client::Client<'a, AxelarExecutableExecuteMsg, ()>> for AxelarExecutableClient<'a> {
-    fn from(client: client::Client<'a, AxelarExecutableExecuteMsg, ()>) -> Self {
-        AxelarExecutableClient { client }
+impl<'a> From<client::Client<'a, AxelarExecutableMsg, ()>> for PayloadExecutor<'a> {
+    fn from(client: client::Client<'a, AxelarExecutableMsg, ()>) -> Self {
+        PayloadExecutor { client }
     }
 }
 
-pub struct AxelarExecutableClient<'a> {
-    client: client::Client<'a, AxelarExecutableExecuteMsg, ()>,
-}
+impl<'a> PayloadExecutor<'a> {
+    pub fn new(deps: Deps<'a>, destination: &str) -> error_stack::Result<Self, address::Error> {
+        let destination = address::validate_cosmwasm_address(deps.api, destination)?;
+        Ok(PayloadExecutor {
+            client: client::Client::new(deps.querier, destination),
+        })
+    }
 
-impl<'a> AxelarExecutableClient<'a> {
     pub fn execute(
         &self,
         cc_id: CrossChainId,
         source_address: Address,
         payload: HexBinary,
     ) -> WasmMsg {
-        self.client
-            .execute(&AxelarExecutableExecuteMsg::Execute(AxelarExecutableMsg {
```

Comment:
we need this wrapping since the app does this wrapping: `ExecuteMsg::Execute(AxelarExecutableMsg)`